include/deal.II/matrix_free/fe_point_evaluation.h

```diff
@@ -476,55 +436,53 @@ class FEPointEvaluation
                     const unsigned int        first_selected_component = 0);
 
   /**
-   * Set up the mapping information for the given cell, e.g., by computing the
-   * Jacobian of the mapping for the given points if gradients of the functions
-   * are requested.
+   * Constructor.
    *
-   * @param[in] cell An iterator to the current cell
+   * @param mapping_info The MappingInfo class is the interface to mapping
+   * computation and storage and is used to precompute mapping.
    *
-   * @param[in] unit_points List of points in the reference locations of the
-   * current cell where the FiniteElement object should be
-   * evaluated/integrated in the evaluate() and integrate() functions.
+   * @param fe The FiniteElement object that is used for the evaluation, which
+   * is typically the same on all cells to be evaluated.
+   *
+   * @param update_flags Specify the quantities to be computed by the mapping
+   * during the call of reinit(). During evaluate() or integrate(), this data
+   * is queried to produce the desired result (e.g., the gradient of a finite
+   * element solution).
+   *
+   * @param first_selected_component For multi-component FiniteElement
+   * objects, this parameter allows to select a range of `n_components`
+   * components starting from this parameter.
+   */
+  FEPointEvaluation(NonMatching::MappingInfo<n_components, dim> &mapping_info,
+                    const FiniteElement<dim> &                   fe,
+                    const UpdateFlags                            update_flags,
+                    const unsigned int first_selected_component = 0);
+
+  /**
+   * Common setup function for both constructors. Does setup for both fast and
+   * slow path.
+   *
+   * @param first_selected_component For multi-component FiniteElement
+   * objects, this parameter allows to select a range of `n_components`
+   * components starting from this parameter.
    */
   void
-  reinit(const typename Triangulation<dim, spacedim>::cell_iterator &cell,
-         const ArrayView<const Point<dim>> &unit_points);
+  setup(const unsigned int first_selected_component);
```

Comment:
Should this function really be in the public interface?

Reply:
No i should not. I changed it to private.